server/routes/createGoal/createGoalsController.ts

```diff
@@ -71,16 +76,22 @@ export default class CreateGoalsController {
       return res.redirectWithErrors(`/plan/${prisonNumber}/goals/create`, errors)
     }
 
+    req.session.createGoalsForm = undefined
+    const createGoalDtos = toCreateGoalDtos(createGoalsForm, prisonId)
+
     try {
-      const createGoalDtos = toCreateGoalDtos(createGoalsForm, prisonId)
       await this.educationAndWorkPlanService.createGoals(prisonNumber, createGoalDtos, req.user.token)
-
-      req.session.createGoalsForm = undefined
-      return res.redirectWithSuccess(`/plan/${prisonNumber}/view/overview`, 'Goals added')
     } catch (e) {
       logger.error(`Error creating goal(s) for prisoner ${prisonNumber}`, e)
       return next(createError(500, `Error creating goal(s) for prisoner ${prisonNumber}. Error: ${e}`))
     }
+
+    await Promise.all(
+      createGoalDtos.map((createGoalDto, idx) =>
+        this.auditService.logCreateGoal(createGoalAuditData(req, idx + 1, createGoalDtos.length)),
+      ),
+    )
+    return res.redirectWithSuccess(`/plan/${prisonNumber}/view/overview`, 'Goals added')
```

Comment:
I probably should've asked this on the other PRs so sorry for bringing it up now.
In the case that the audit promise fails we'd show an error to the user but we would've actually created the goals. Maybe it should just log and not return an error?

Reply:
That is a good point, and you are right in that it affects the recent PRs in this story, but also the existing calls to the audit service (all page view attempts and all page views)
Leave it with me - I'll add something to address this 👍

Reply:
Sorted 👍 
As it happens `HmppsAuditClient` already has us covered for this, with it's optional `throwOnError` parameter on the `sendMessage` method (and all of this is already unit tested 👍 )
```
  async sendMessage(event: AuditEvent, throwOnError: boolean = true) {
    if (!this.enabled) return null

    const sqsMessage: SqsMessage = {
      ...event,
      details: JSON.stringify(event.details),
      service: this.serviceName,
      when: new Date().toISOString(),
    }

    try {
      const messageResponse = await this.sqsClient.send(
        new SendMessageCommand({ MessageBody: JSON.stringify(sqsMessage), QueueUrl: this.queueUrl }),
      )

      logger.info(`HMPPS Audit SQS message sent (${messageResponse.MessageId})`)

      return messageResponse
    } catch (error) {
      logger.error('Error sending HMPPS Audit SQS message, ', error)
      if (throwOnError) throw error
    }
    return null
  }
``` 

So we just need to make sure we pass `throwOnError` as `false` in our various methods in `AuditService` 👍 

I also took the opportunity to remove the `await` keywords to all the places we call the methods in `AuditService` because it's async fire'n'forget - the controllers don't need to wait for the response 👍